app/models/concerns/has_managed_assets.rb

```diff
@@ -0,0 +1,20 @@
+module HasManagedAssets
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def managed_image(attr, options={})
+      belongs_to :"#{attr}_association", foreign_key: :"#{attr}_asset_id", class_name: "Image"
+      define_method attr do
+        send("#{attr}_association").try(:data)
+      end
+    end
+
+    def managed_document(attr, options={})
+      belongs_to :"#{attr}_association", foreign_key: :"#{attr}_asset_id", class_name: "Document"
+      define_method attr do
+        send("#{attr}_association").try(:data)
+      end
+    end
+  end
```

Comment:
@dbaines In case you're interested, I've changed these to use `belongs_to` associations. This should end up being more performant when we're listing, say, 50 things with managed assets, since we can now do 

```
KidHero.includes(:managed_image_association).map(&:managed_image)
```

and it'll do two database queries, rather than 51 queries. There are probably other benefits to using an association as well!